src/repository/opamRepositoryBackend.ml

```diff
@@ -171,8 +171,20 @@ let get_diff parent_dir dir1 dir2 =
     log "Internal diff (empty) done in %.2fs." (chrono ());
     None
   | diffs ->
-    log "Internal diff (non-empty) done in %.2fs." (chrono ());
+    let changed_files =
+      List.map (fun diff ->
+          let file = match diff.Patch.operation with
+            | Patch.Edit (file, _) -> file
+            | Patch.Delete file -> file
+            | Patch.Create file -> file
+            | Patch.Git_ext (file, _, _) -> file
```

Comment:
i think you want to keep the whole `diff.Patch.operation` around. Otherwise you won't know when a file has been moved

Reply:
i think this also warrants a test. I don't think we have a test for files being moved around, empty files being deleted and empty files being created in a git repository.

Reply:
> first we have to parse the patch file produced by git diff

this should already be done. The parse is done in `OpamRepository.apply_repo_update` for all backends, it would be nice to also avoid duplication with data we already have in the case of the HTTP/local backend.

Reply:
> The parse is done in done in OpamRepository.apply_repo_update... avoid duplication in the case of the HTTP/local backend.

In `OpamRepositoryBackend` the patch file is created after the we compute `Patch.diff` for each file (diffs that I currently reuse). And for VCS it's the other way around (`Patch.parse` produces the diffs). So I feel like I should reuse the the info from `OpamRepository.apply_repo_update` (by propagating it from `OpamSystem.patch`) and keep the HTTP/local backends as is.